lib/active_job/queue_adapters/pubsub_adapter.rb

```diff
@@ -7,15 +7,32 @@ class PubsubAdapter
       #
       # @param [ActiveJob::Base] job The job to be performed.
       def enqueue(job)
-        raise(NotImplementedError)
+        puts("\n [PubsubAdapter][enqueue]: #{job.inspect}")
+        publish_job(job)
       end
 
       # Enqueue a job to be performed at a certain time.
       #
       # @param [ActiveJob::Base] job The job to be performed.
       # @param [Float] timestamp The time to perform the job.
       def enqueue_at(job, timestamp)
-        raise(NotImplementedError)
+        puts("\n [PubsubAdapter][enqueue_at]: #{job.inspect} timestamp: #{timestamp}")
+        publish_job(job, timestamp)
+      end
+
+       # push job to the topic
+      def publish_job(job, timestamp = 0)
+        puts("\n [PubsubAdapter][publish_job]: #{job.inspect}")
```

Comment:
Same, could this be solved using instrumentation?

Reply:
Yes, I added the commit https://github.com/skysemilla/kisi-api-challenge/pull/1/commits/8425060cc4cd4472de51ee68f8080f3ed8d6d764 to implement the active support notification to instrument and subscribe to an event.